app/scripts/controllers/transactions/tx-state-manager.js

```diff
@@ -278,9 +272,9 @@ export default class TransactionStateManager extends EventEmitter {
     const txsToDelete = transactions
       .reverse()
       .filter((tx) => {
-        const { nonce, from } = tx.txParams;
-        const { chainId, metamaskNetworkId, status } = tx;
-        const key = `${nonce}-${chainId ?? metamaskNetworkId}-${from}`;
+        const { chainId, status, txParams } = tx;
+        const txChainId = chainId ?? txParams.chainId;
```

Comment:
This fallback isn't required as both are set using the value from the network state, plus the `chainId` is stripped from the `txParams` during normalisation.